chore: clean up access control utils

zlwaterfield opened the pull request:
## Changes

No function changes, just moving utilts into a single file and adding comments 

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

zlwaterfield committed:
clean up access control utils

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield committed:
more cleanup

zlwaterfield committed:
Merge branch 'zach/access-control-util-cleanup' of github.com:PostHog/posthog into zach/access-control-util-cleanup

zlwaterfield committed:
Update accessControlUtils.ts

zlwaterfield committed:
import fixes

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield committed:
More types fixes

zlwaterfield committed:
Merge branch 'zach/access-control-util-cleanup' of github.com:PostHog/posthog into zach/access-control-util-cleanup

github-actions[bot] committed:
Update E2E screenshots for Playwright

github-actions[bot] committed:
Update E2E screenshots for Playwright

zlwaterfield committed:
Merge branch 'master' into zach/access-control-util-cleanup

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

zlwaterfield committed:
Refactor access control imports for surveys and sidepanel

Moved imports of resourceTypeToString and userHasAccess from 'lib/components/AccessControlAction' to 'accessControlUtils' in the sidepanel directory. This improves code organization and ensures access control utilities are imported from a more appropriate location.

a-lider reviewed (approved):
Thanks for cleaning this up!
General question - should these access control utils live somewhere higher level? This is quite long: `import { accessLevelSatisfied } from '~/layout/navigation-3000/sidepanel/panels/access_control/accessControlUtils'`
maybe in the `~/lib/utils/`?

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

zlwaterfield committed:
Move accessControlUtils to lib/utils and update imports

Relocated accessControlUtils.ts from the access_control panel directory to lib/utils for broader utility. Updated all relevant imports across the codebase to reference the new location.

zlwaterfield committed:
Merge branch 'zach/access-control-util-cleanup' of github.com:PostHog/posthog into zach/access-control-util-cleanup

yasen-posthog commented:
> Thanks for cleaning this up! General question - should these access control utils live somewhere higher level? This is quite long: `import { accessLevelSatisfied } from '~/layout/navigation-3000/sidepanel/panels/access_control/accessControlUtils'` maybe in the `~/lib/utils/`?

I'd agree with this tbh, access_control scoped under sidepanel is weird.

zlwaterfield commented:
Done!
